spec/abci/abci.md

```diff
@@ -5,27 +5,32 @@ title: Method and Types
 
 # Methods and Types
 
-## Overview
-
-The ABCI message types are defined in a [protobuf
-file](https://github.com/tendermint/tendermint/blob/master/proto/tendermint/abci/types.proto).
-
-ABCI methods are split across four separate ABCI _connections_:
-
-- Consensus connection: `InitChain`, `BeginBlock`, `DeliverTx`, `EndBlock`, `Commit`
-- Mempool connection: `CheckTx`
-- Info connection: `Info`, `Query`
-- Snapshot connection: `ListSnapshots`, `LoadSnapshotChunk`, `OfferSnapshot`, `ApplySnapshotChunk`
-
-The consensus connection is driven by a consensus protocol and is responsible
-for block execution.
-
-The mempool connection is for validating new transactions, before they're
-shared or included in a block.
-
-The info connection is for initialization and for queries from the user.
-
-The snapshot connection is for serving and restoring [state sync snapshots](apps.md#state-sync).
+## Connections
+
+ABCI applications can run either within the same _process_ as the Tendermint 
+state-machine replication engine _or_ as a separate process from the state-machine
+replication engine. When run within the same process, Tendermint will call the ABCI
+application methods directly as typical Go method calls. When Tendermint and the
+ABCI application are run as separate processes, Tendermint opens a set of connections
+to the application.
+
+When Tendermint and the application are run as separate processes, Tendermint opens
+four connections for ABCI methods. The connections each handle a subset of the ABCI
+method calls. They are defined as follows:
+
+#### **Consensus** connection
+* Handles the `InitChain`, `BeginBlock`, `DeliverTx`, `EndBlock`, and `Commit` method
```

Comment:
I'd suggest putting the "summary" explanation first, and the list of methods second, in each case.

Reply:
done.